Potential changes to Rules for AI Triage Service (#9964)
Azure/azure-sdk-tools: tools/github-event-processor/RULES.md

### AI label service

> This is a REST API that evaluates the content of an issue and attempts to predict a set of labels and provide a relevant comment. If predictions is made with an extreme level of confidence, the service will return one service label, one category label and a solution comment. In this case an `issue-addressed` is added to the issue to close it. If a prediction is made with a reasonable level of confidence, the service will return one service label, one category label, and a suggestion comment.  If confidence thresholds were not met, labels nor a comment are provided. The AI label service can only a response with labels for [officially supported repositories]

Reviewer: ```suggestion
This is a REST API that evaluates the content of an issue and attempts to predict a set of labels and provide a relevant answer. If predictions are made with a high degree of confidence, the service will return one service label, one category label and a solution comment. In this case an `issue-addressed` is added to the issue to close it. If a prediction is made with a reasonable level of confidence, the service will return one service label, one category label, and a suggestion comment.  If confidence thresholds were not met, labels nor a comment are provided. The AI label service can only a response with labels for [officially supported repositories]
```

#### Example responses

> _**Only Labels Predicted**_

Reviewer: ```suggestion
_**Only labels predicted**_
```

```json
{

>   "labels" : [

Reviewer: I'd suggest that we break this out into the 4 possible states with examples for each:

- Manual triage needed.   (empty labels, no suggestions, no solution)
- Label prediction made for triage.  (labels populated, no suggestion, no solution)
- Label prediction with suggestion (labels populated, suggestion populated, no solution)
- Label prediction with solution (labels populated, no suggestion, solution populated)

Author: For suggestion/solution example I just provided the first bit. Not sure if you would want a complete example.

    "Storage",
    "Client"
  ],  

>   "suggestion" : string | null,  
>   "solution" : string | null  

Reviewer: ```suggestion
  "suggestion" : null,  
  "solution" : null  
```

```json
{
  "labels": [],  
  "answer" : null,  

>   "answerType" : "none"  

Reviewer: Since we're not doing an enum, should we go with `null` here or use the "none" string?    I'm on the fence and don't feel strongly either way.

Author: think null would make more sense for consistency's sake (since answer is also null when nothings there).

    IF "needs-team-triage" is not being added to the issue
         - Add "needs-team-attention" label to the issue

>     IF suggestions is populated

Reviewer: ```suggestion
    IF suggestion is populated
```
